### PR TITLE
Slight performance enchancement for crosshair

### PR DESCRIPTION
**Pull request by rjmcneill**
Relates to #696 
Relates to #746

### CODE REVIEW

**Comment by jdhodges** [comment]
Seems to be a slight issue. Try reloading the page then changing the series to area and back to candlestick again.

**Comment by rjmcneill** [comment]
@jdhodges should be resolved now :)

File: src/assets/js/app.js [hunk 1]
```diff
@@ -162,6 +162,14 @@ export default function() {
         }
     }
 
+    function renderLegend() {
```
Comment (rjmcneill) [inline src/assets/js/app.js]: Would `renderCrosshairChange` maybe be a more appropriate name for this?

File: src/assets/js/chart/primary.js [hunk 2]
```diff
@@ -139,8 +139,16 @@ export default function() {
         model.selectorsChanged = false;
     }
 
-    function bandCrosshair(data) {
-        var width = currentSeries.option.width(data);
+    function bandCrosshair(model) {
+        var mapFunction = function(d, i) { return xScale(xValue(d, i)); };
+        var barWidth = currentSeries.option.barWidth();
+        var xValue = currentSeries.option.xValue();
+        var yValue = function(d) { return d.volume; };
```
Comment (rjmcneill) [inline src/assets/js/chart/primary.js]: `d.volume` change this

File: src/assets/js/chart/primary.js [hunk 3]
```diff
@@ -139,8 +139,16 @@ export default function() {
...
+        var y0Value = d3.functor(0);
+        var x0Value = d3.functor(0);
+
+        var filteredData = model.data.filter(fc.util.fn.defined(x0Value, y0Value, xValue, yValue));
+        var width = barWidth(filteredData.map(mapFunction));
```
Comment (jdhodges) [inline src/assets/js/chart/primary.js]: Don't think we need all of this, how about just this:
```
var xValue = currentSeries.option.xValue();
var mapFunction = function(d, i) { return xScale(xValue(d, i)); };
var barWidth = currentSeries.option.barWidth();

var width = barWidth(model.data.map(mapFunction));
```

**Pull request closed without merging**